pkg/sdk/warehouse.go

```diff
@@ -135,81 +134,99 @@ type WarehouseAlterOptions struct {
-	WarehouseSize                   *WarehouseSize `ddl:"parameter,single_quotes" db:"WAREHOUSE_SIZE"`
-	WaitForCompletion               *bool          `ddl:"parameter" db:"WAIT_FOR_COMPLETION"`
-	MaxClusterCount                 *uint8         `ddl:"parameter" db:"MAX_CLUSTER_COUNT"`
-	MinClusterCount                 *uint8         `ddl:"parameter" db:"MIN_CLUSTER_COUNT"`
-	ScalingPolicy                   *ScalingPolicy `ddl:"parameter,single_quotes" db:"SCALING_POLICY"`
-	AutoSuspend                     *uint          `ddl:"parameter" db:"AUTO_SUSPEND"`
-	AutoResume                      *bool          `ddl:"parameter" db:"AUTO_RESUME"`
-	ResourceMonitor                 *string        `ddl:"parameter,double_quotes" db:"RESOURCE_MONITOR"`
-	Comment                         *string        `ddl:"parameter,single_quotes" db:"COMMENT"`
-	EnableQueryAcceleration         *bool          `ddl:"parameter" db:"ENABLE_QUERY_ACCELERATION"`
-	QueryAccelerationMaxScaleFactor *uint8         `ddl:"parameter" db:"QUERY_ACCELERATION_MAX_SCALE_FACTOR"`
+	WarehouseType                   *WarehouseType          `ddl:"parameter,single_quotes" db:"WAREHOUSE_TYPE"`
+	WarehouseSize                   *WarehouseSize          `ddl:"parameter,single_quotes" db:"WAREHOUSE_SIZE"`
+	WaitForCompletion               *bool                   `ddl:"parameter" db:"WAIT_FOR_COMPLETION"`
+	MaxClusterCount                 *int                    `ddl:"parameter" db:"MAX_CLUSTER_COUNT"`
+	MinClusterCount                 *int                    `ddl:"parameter" db:"MIN_CLUSTER_COUNT"`
+	ScalingPolicy                   *ScalingPolicy          `ddl:"parameter,single_quotes" db:"SCALING_POLICY"`
+	AutoSuspend                     *int                    `ddl:"parameter" db:"AUTO_SUSPEND"`
+	AutoResume                      *bool                   `ddl:"parameter" db:"AUTO_RESUME"`
+	ResourceMonitor                 AccountObjectIdentifier `ddl:"identifier,equals" db:"RESOURCE_MONITOR"`
+	Comment                         *string                 `ddl:"parameter,single_quotes" db:"COMMENT"`
+	EnableQueryAcceleration         *bool                   `ddl:"parameter" db:"ENABLE_QUERY_ACCELERATION"`
+	QueryAccelerationMaxScaleFactor *int                    `ddl:"parameter" db:"QUERY_ACCELERATION_MAX_SCALE_FACTOR"`
 
 	// Object params
-	MaxConcurrencyLevel             *uint `ddl:"parameter" db:"MAX_CONCURRENCY_LEVEL"`
-	StatementQueuedTimeoutInSeconds *uint `ddl:"parameter" db:"STATEMENT_QUEUED_TIMEOUT_IN_SECONDS"`
-	StatementTimeoutInSeconds       *uint `ddl:"parameter" db:"STATEMENT_TIMEOUT_IN_SECONDS"`
+	MaxConcurrencyLevel             *int `ddl:"parameter" db:"MAX_CONCURRENCY_LEVEL"`
+	StatementQueuedTimeoutInSeconds *int `ddl:"parameter" db:"STATEMENT_QUEUED_TIMEOUT_IN_SECONDS"`
+	StatementTimeoutInSeconds       *int `ddl:"parameter" db:"STATEMENT_TIMEOUT_IN_SECONDS"`
+
+	Tag []TagAssociation `ddl:"keyword" db:"TAG"`
```

Comment:
Pretty sure I tried this last week but turns out tags can't be set at the same time as normal parameters, it's either `SET param = 'val';` or `SET TAG db.s.tag = 'val';`, we can't mix both. Same for unset.
Hence why I created a separate field in `WarehouseAlterOptions`.

Reply:
i can do validation to ensure that the user only sets one or the other but i prefer to keep it this way